### PR TITLE
Add AbstractTransformer.ALLOW_CYCLIC_PUBLISHING

### PR DESCRIPTION
**Pull request by aledsage**
If set to true, then allows the transformer to read and publish on the 
same sensor. This is needed for some use-cases in 
brooklyncentral/advanced-networking.

### CODE REVIEW

**Comment by aledsage** [comment]
See https://github.com/brooklyncentral/advanced-networking/pull/99 for an example usage of this.

File: core/src/test/java/org/apache/brooklyn/enricher/stock/TransformingEnricherTest.java [hunk 1]
```diff
@@ -225,17 +226,38 @@ public void testInfersTargetSensorSameAsSource() throws Exception {
...
+                        if (input != null && input < 10) {
+                            return input + 1;
+                        } else {
+                            return Entities.UNCHANGED;
+                        }
+                    }}));
+
+        app.sensors().set(intSensorA, 3);
+
+        EntityAsserts.assertAttributeEqualsEventually(app, intSensorA, 10);
```
Comment (nakomis) [inline core/src/test/java/org/apache/brooklyn/enricher/stock/TransformingEnricherTest.java]: Follow up with `EntityAsserts.assertAttributeEqualsContinually` to ensure value remains at 10

**Comment by nakomis** [comment]
One comment, but other than that, LGTM

**Comment by aledsage** [comment]
Thanks @nakomis - merging.

**Pull request merged**